cityofaustin/atd-moped
# Edit component attributes form briefly shows previous values on save

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/11612

This PR updates the component attributes edit form to fix an issue where the previous values briefly flashed in the UI when saving. I included a diagram of the refactor but it comes down to updating the UI through the new edits through the same channel that the initial data comes through. This PR also pulls out all of the data prep for the form fields out of a custom hook and into individual helpers.

## Testing
**URL to test:** 

Local

**Steps to test:**
1. Through a project's Map tab, create a signal component (Signal - PHB or Signal - Traffic) and another non-signal component with all parts of the component form filled in
2. Edit these attributes and make sure the previous values don't show as the modal is closing
3. Try creating other components with different combinations of attributes filled in the form
4. Edit the components to make sure:
   - the form will not let you submit without selecting a component type
   - the form will not let you submit without selecting a signal type if you choose a signal component
   - all of the fields that you added when creating a component are shown in the form when you edit a component
   - the submit button is disabled until you make a change from the initial value of a field

![change](https://github.com/cityofaustin/atd-moped/assets/37249039/b143f7ae-dc6c-42cc-8237-c7b0488e1351)

---
#### Ship list
